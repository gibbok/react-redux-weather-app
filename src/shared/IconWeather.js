import React from 'react'
import getIconClassName from '../app/weatherIcons'

const IconWeather = ({code}) => {
  const iconClassName = getIconClassName(code)
  return (
    <i className={iconClassName} />
  )
}

export default IconWeather
