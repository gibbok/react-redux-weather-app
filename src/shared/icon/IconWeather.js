
// @flow
import * as React from 'react'
import getIconClassName from '../../app/weatherIcons'

/* eslint-disable no-undef */
type PropsType = {
  +code?: number,
}
/* eslint-enable no-undef */

const IconWeather:React.StatelessFunctionalComponent<any> = ({code}:PropsType) => {
  if (code === undefined) {
    return null
  }
  const iconClassName = getIconClassName(code)
  return (
    <i className={iconClassName} />
  )
}

export default IconWeather
