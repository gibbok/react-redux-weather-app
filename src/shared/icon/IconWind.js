// @flow
import * as React from 'react'

/* eslint-disable no-undef */
type PropsType = {
  +degree: number,
}
/* eslint-enable no-undef */

const IconWind:React.StatelessFunctionalComponent<PropsType> = ({degree}:PropsType):React.Element<any> => {
  const degreeAdjusted = -90 + degree
  const style = {
    display: 'inline-block',
    transform: `rotate(${degreeAdjusted}deg)`
  }
  return (
    <span style={style}>&#10148;</span>
  )
}

export default IconWind
