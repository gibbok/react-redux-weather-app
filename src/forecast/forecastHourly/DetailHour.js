// @flow
import * as React from 'react'
import moment from 'moment'
import IconWeather from '../../shared/icon/IconWeather'
import IconWind from '../../shared/icon/IconWind'
import { withStyles, Typography, Divider } from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +momentId: string,
  +timespan: number,
  +temp: number,
  +weatherIconCode: number,
  +weatherDescription: string,
  +windDegree: number,
  +windSpeed: number,
  +humidityUnit: '%',
  +humidityValue: number,
  +classes: Object
}
/* eslint-enable no-undef */

const styles = theme => {
  const spaceUnit = theme.spacing.unit
  return ({
    root: {
      width: 112
    },
    icon: {
      marginBottom: spaceUnit * 0.5
    },
    temperature: {
      marginBottom: spaceUnit * 0.5
    },
    description: {
      marginBottom: spaceUnit
    },
    wind: {},
    humidity: {
      marginBottom: spaceUnit * 1.25
    },
    hour: {}
  })
}

const DetailHour:React.StatelessFunctionalComponent<PropsType> = ({ momentId, timespan, temp, weatherIconCode, weatherDescription, windDegree, windSpeed, humidityUnit, humidityValue, classes }:PropsType):React.Element<any> => {
  const hourFormat = moment.unix(timespan).format('h A')
  const tempFormat = Math.round(temp)
  const windSpeedFormat = Math.round(windSpeed)
  return (
    <div className={classes.root}>
      <div className={classes.icon}>
        <Typography type='headline'>
          <IconWeather code={weatherIconCode} />
        </Typography>
      </div>
      <div className={classes.temperature}>
        <Typography type='title'>
          {tempFormat} &#176;
        </Typography>
      </div>
      <div className={classes.description}>
        <Typography type='subheading'>
          {weatherDescription}
        </Typography>
      </div>
      <div className={classes.wind}>
        <Typography type='body1'>
          Wind <IconWind degree={windDegree} /> {windSpeedFormat} mps
        </Typography>
      </div>
      <div className={classes.humidity}>
        <Typography type='body1'>
          Humidity {humidityValue} {humidityUnit}
        </Typography>
      </div>
      <div className={classes.hour}>
        <Divider />
        <Typography type='subheading'>
          <strong>{hourFormat}</strong>
        </Typography>
      </div>
    </div>
  )
}

export default withStyles(styles)(DetailHour)
