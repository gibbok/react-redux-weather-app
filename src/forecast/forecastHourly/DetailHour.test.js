// @flow
import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import DetailHour from './DetailHour'
import IconWeather from '../../shared/icon/IconWeather'
import IconWind from '../../shared/icon/IconWind'

describe('<DetailHour />', () => {
  it('should render', () => {
    const wrapper = shallow(
      <DetailHour
        momentId={'2017:8:13:3'}
        timespan={1505260800}
        temp={21.57}
        weatherIconCode={803}
        weatherDescription={'Clouds'}
        windDegree={227.007}
        windSpeed={2.56}
        humidityUnit={'%'}
        humidityValue={63}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should contain one IconWeather component', () => {
    const wrapper = shallow(
      <DetailHour
        momentId={'2017:8:13:3'}
        timespan={1505260800}
        temp={21.57}
        weatherIconCode={803}
        weatherDescription={'Clouds'}
        windDegree={227.007}
        windSpeed={2.56}
        humidityUnit={'%'}
        humidityValue={63}
        />
      )
    expect(wrapper.find(IconWeather).length).toBe(1)
  })

  it('should contain one IconWind component', () => {
    const wrapper = shallow(
      <DetailHour
        momentId={'2017:8:13:3'}
        timespan={1505260800}
        temp={21.57}
        weatherIconCode={803}
        weatherDescription={'Clouds'}
        windDegree={227.007}
        windSpeed={2.56}
        humidityUnit={'%'}
        humidityValue={63}
        />
      )
    expect(wrapper.find(IconWind).length).toBe(1)
  })
})
