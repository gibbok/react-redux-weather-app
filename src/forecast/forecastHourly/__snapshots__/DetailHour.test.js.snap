// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailHour /> should render 1`] = `
<div
  className="DetailHour-root-1"
>
  <div>
    <withStyles(Typography)
      type="title"
    >
      <IconWeather
        code={803}
      />
    </withStyles(Typography)>
  </div>
  <div>
    <withStyles(Typography)
      type="subheading"
    >
      22
       Â°
    </withStyles(Typography)>
  </div>
  <div>
    <withStyles(Typography)
      type="subheading"
    >
      Clouds
    </withStyles(Typography)>
  </div>
  <div>
    <withStyles(Typography)
      type="body1"
    >
      Wind 
      <IconWind
        degree={227.007}
      />
       
      3
       mps
    </withStyles(Typography)>
  </div>
  <div>
    <withStyles(Typography)
      type="body1"
    >
      Humidity 
      63
       
      %
    </withStyles(Typography)>
  </div>
  <div>
    <withStyles(Divider) />
    <withStyles(Typography)
      type="subheading"
    >
      <strong>
        2 AM
      </strong>
    </withStyles(Typography)>
  </div>
</div>
`;
